### PR TITLE
avoid compilation errors on OSX(tiles solver)

### PR DESCRIPTION
**Pull request by spinute**
I wanted to try tiles solvers on OSX, however, I saw some errors during build through "make tiles".

I haven't understood the codes yet, but I tried fixing compilation error at least on my environment(OSX10.11.6), and I want to share it with other OSX users.

I will use Linux environment for serious benchmarks, but it is excellent to run this solver in my usual environment, and hence if you have any suggestions for my way of fixes in this commit, I would appreciate if you could tell me about that.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,7 +21,7 @@ CXXFLAGS:=$(FLAGS) -std=c++0x
 
 CFLAGS:=$(FLAGS) -std=c99
 
-LDFLAGS:=$(FLAGS) -std=c++0x -lrt
```
Comment (eaburns) [inline Makefile]: Does removing -lrt break other systems?

File: tiles/mdist.cc [hunk 2]
```diff
@@ -30,7 +30,12 @@ void TilesMdist::initmd() {
 		for (int i = 0; i < Ntiles; i++) {
 			unsigned int r = i / Width;
 			unsigned int c = i % Width;
-			md[t][i] = abs(r - row) + abs(c - col);
+			/* XXX: tiles/mdist.cc:33:15: error: taking the absolute value of unsigned type 'unsigned int'
+			 *  has no effect [-Werror,-Wabsolute-value]
+			 *  md[t][i] = abs(r - row) + abs(c - col); */
+			//md[t][i] = abs(r - row) + abs(c - col);
+
```
Comment (eaburns) [inline tiles/mdist.cc]: Thanks for finding this bug. Please remove the comment.

**Comment by eaburns** [comment]
Thanks. This looks fine. Just a few comments.

File: Makefile (hunk 1 above)
Comment (spinute) [inline Makefile]: Sorry for my lazyness:(
I added branch for OSX.

File: tiles/Make.inc [hunk 3]
```diff
@@ -17,6 +17,13 @@ TILESSRC:=\
 
 CLEAN+=$(TILESSRC:.cc=.o)
 
+OS := $(shell uname)
```
Comment (spinute) [inline tiles/Make.inc]: I also added a branch to revert to make static library on other environments than OSX.

File: tiles/mdist.cc [hunk 4]
```diff
@@ -30,7 +30,8 @@ void TilesMdist::initmd() {
 		for (int i = 0; i < Ntiles; i++) {
 			unsigned int r = i / Width;
 			unsigned int c = i % Width;
-			md[t][i] = abs(r - row) + abs(c - col);
+
+			md[t][i] = (r - row) + (c - col);
 		}
```
Comment (spinute) [inline tiles/mdist.cc]: I removed comments here.

**Comment by eaburns** [comment]
Thanks a lot! Looks good. Squash those two commits and I'll pull it.

File: tiles/mdist.cc [hunk 5]
```diff
@@ -25,11 +25,12 @@ TilesMdist::State TilesMdist::initialstate() {
...
-		unsigned int row = goalpos[t] / Width;
-		unsigned int col = goalpos[t] % Width;
+		int row = goalpos[t] / Width;
+		int col = goalpos[t] % Width;
 		for (int i = 0; i < Ntiles; i++) {
-			unsigned int r = i / Width;
-			unsigned int c = i % Width;
+			int r = i / Width;
+			int c = i % Width;
+
```
Comment (spinute) [inline tiles/mdist.cc]: I didn't read the code not enough, and I did a hasty decision.
My previous fix was broken as a calculation of Manhattan distance, and I fixed it to use abs again.

**Comment by spinute** [comment]
Thank you for your comments.
I rebased these commits into one.

**Comment by eaburns** [comment]
Thanks.

**Pull request merged**